### PR TITLE
Implements TypeScript types for path and order book methods

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR creates new TypeScript types for all rippled path and order book methods (https://xrpl.org/path-and-order-book-methods.html), both the request shapes and response shapes.

### Context of Change

xrpl.js v2.0 models

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Test Plan

This PR doesn't change any existing code, so tests are fine.

### CODE REVIEW

File: src/models/methods/pathFind.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+export interface PathFindResponse extends BaseResponse {
+  result: {
+      alternatives: PathOption[]
+      destination_account: string
+      destination_amount: Amount
+      source_account: string
+      full_reply: boolean
+      id: number | string
```
Comment (JST5000) [inline src/models/methods/pathFind.ts:40]: Will we need to update the documentation when 2.0 comes out to limit id's to numbers/strings? (It seems websocket supports any unique object, but we have a convention that is partially applied)

File: src/models/methods/ripplePathFind.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    issuer?: string
+  }
+
+export interface RipplePathFindRequest extends BaseRequest {
+    command: "ripple_path_find"
+    source_account: string
+    destination_account: string
+    destination_amount: Amount
+    send_max?: Amount
+    source_currencies: SourceCurrencyAmount
```
Comment (JST5000) [inline src/models/methods/ripplePathFind.ts:15]: It seems like source_currencies should be optional - https://xrpl.org/ripple_path_find.html#ripple_path_find

**Review by JST5000 — COMMENTED** [review]
Just one line seems like it should be optional, other than that good to go!

File: src/models/methods/pathFind.ts (hunk 1 above)
Comment (natenichols) [reply src/models/methods/pathFind.ts:40]: `id` should be optional IIRC, and a member of `BaseRequest`/`BaseResponse`

```ts
id?: number | string
```

I think `id` can be anything (so we could type it as `any`), and rippled just echoes it back to you, but I dont really see the harm in restricting it to a `number | string`.

**Review by natenichols — APPROVED** [review]
Left one comment, but nothing I really felt should hold up my approval.

File: src/models/methods/ripplePathFind.ts (hunk 2 above)
Comment (mvadari) [reply src/models/methods/ripplePathFind.ts:15]: Good catch, thanks

File: src/models/methods/pathFind.ts (hunk 1 above)
Comment (JST5000) [reply src/models/methods/pathFind.ts]: @natenichols in this specific case id is echoed in the response, in addition to the id? at the top level in BaseRequest/BaseResponse

**Review by JST5000 — APPROVED** [review]

**Pull request merged**